python/ray/tune/impl/tuner_internal.py

```diff
@@ -143,9 +135,24 @@ def __init__(
         # without allowing for checkpointing tuner and trainable.
         # Thus this has to happen before tune.run() so that we can have something
         # to restore from.
-        experiment_checkpoint_path = Path(self._local_experiment_dir, _TUNER_PKL)
-        with open(experiment_checkpoint_path, "wb") as fp:
-            pickle.dump(self.__getstate__(), fp)
+        self._run_config.name = (
+            self._run_config.name
+            or StorageContext.get_experiment_dir_name(self.converted_trainable)
+        )
+        storage = StorageContext(
```

Comment:
Should we be trying to instantiate this `StorageContext` only in one place and then pass it around?

Reply:
This is hard to do since we'd need to pass the context through from `BaseTrainer.fit` to the public `Tuner` interface.

One alternative we discussed is having a global storage context, but that'd also require a `get_or_create_storage_context` logic to account for users coming in through any of the 3 entrypoints (tuner, trainer, tune.run).

I will the usage is just to access the path and the filesystem so that we don't re-implement that logic.